packages/datacommons-mcp/datacommons_mcp/data_models/settings.py

```diff
@@ -136,5 +140,15 @@ def compute_api_base_url(self) -> "CustomDCSettings":
         return self
 
 
+def _parse_list_like_parameter(v: Any) -> list[str] | None:  # noqa: ANN401
+    """Parse comma-separated string into list of strings."""
+    if not isinstance(v, str) or not v.strip():
+        return None
+    # Split by comma and strip whitespace from each item
+    items = [item.strip() for item in v.split(",")]
+    # Filter out empty items
+    return [item for item in items if item]
```

Comment:
@keyurva this changed the logic so that if a list is passed, it will be replaced by `None`. Previously a list would be treated as a list, and it would be list-like strings that would get parsed into a list.
I'll open a quick PR to address this.